packages/backend-lib/src/users.ts

```diff
@@ -81,25 +113,24 @@ function buildUserIdQueries({
     userIdsCondition = Prisma.sql`1=1`;
   }
 
-  const segmentIdCondition = segmentId
-    ? Prisma.sql`"segmentId" = CAST(${segmentId} AS UUID)`
+  const segmentIdCondition = segmentFilter
+    ? Prisma.sql`("segmentId" IN (${Prisma.join(segmentFilter.map((segmentId) => Prisma.sql`${segmentId}::uuid`))}))`
```

Comment:
At some point we'll want to re-implement this method to query clickhouse instead of postgres, but that's not necessary for the time being.